src/test/java/com/microsoft/sqlserver/jdbc/parametermetadata/ParameterMetaDataCacheTest.java

```diff
@@ -0,0 +1,248 @@
+/*
+ * Microsoft JDBC Driver for SQL Server Copyright(c) Microsoft Corporation All rights reserved. This program is made
+ * available under the terms of the MIT License. See the LICENSE file in the project root for more information.
+ */
+package com.microsoft.sqlserver.jdbc.parametermetadata;
+
+import static org.junit.Assert.fail;
+import static org.junit.jupiter.api.Assertions.assertSame;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.sql.Connection;
+import java.sql.ParameterMetaData;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.HashMap;
+import java.util.Map;
+
+import com.microsoft.sqlserver.jdbc.*;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.platform.runner.JUnitPlatform;
+import org.junit.runner.RunWith;
+
+import com.microsoft.sqlserver.testframework.AbstractSQLGenerator;
+import com.microsoft.sqlserver.testframework.AbstractTest;
+import com.microsoft.sqlserver.testframework.Constants;
+
+
+/**
+ * Tests for caching parameter metadata in sp_describe_parameter_encryption calls
+ */
+@RunWith(JUnitPlatform.class)
+public class ParameterMetaDataCacheTest extends AbstractTest {
+    private static final String firstTable = "firstTable";
+    private static final String secondTable = "secondTable";
+    private final String cmkName = "my_cmk";
+    private final String cekName = "my_cek_1";
+    private final String cekNameAlt = "my_cek_2";
+
+    @BeforeAll
+    public static void setupTests() throws Exception {
+        setConnection();
+    }
+
+    public void tableSetup() throws SQLServerException {
+        try (Statement stmt = connection.createStatement()) {
+            TestUtils.dropTableIfExists(firstTable, stmt);
+            TestUtils.dropTableIfExists(secondTable, stmt);
+
+            dropCEK(cekName);
+            dropCMK(cmkName);
+            createCMK(cmkName, keyIDs[0]);
+            createCEK(cekName, cmkName, setupKeyStoreProvider(), keyIDs[0]);
+
+            createTable(cekName, firstTable, "firstColumn", "secondColumn");
+            createTable(cekName, secondTable, "firstColumn", "secondColumn");
+        } catch (SQLException e) {
+            fail(e.getMessage());
+        }
+
+    }
+
+    /**
+     * 
+     * Tests caching of parameter metadata by running a query to be cached, another to replace parameter information,
+     * then the first again to measure the difference in time between the two runs.
+     * 
+     * @throws SQLServerException
+     */
+    @Test
+    @Tag(Constants.xAzureSQLDW)

```

Comment:
why are we skipping DW everywhere? doesn't this work for DW? we should excludeSQLv11, 12, 14 tho since AE is only supposed after that